### ISSUE TITLE
Ubuntu 16.04 - Build stopped - subcommand failed

### ISSUE THREAD

**Issue by aliireza** [scylladb/scylla#3048]
Based on the instructions on HACKING.md, I did the following:

$ git clone https://github.com/scylladb/scylla
$ cd scylla
$ git submodule update --init --recursive
$ sudo ./install_dependencies.sh
.$ /configure.py --mode=all --dpdk-target /opt/dpdk/x86_64-native-linuxapp-gcc

666.development-0.20171206.aeb6ebc
Notice: -fsanitize=vptr is broken, disabling; some debug mode tests are bypassed.
[2/2] GEN build/release/seastar.pc

$ ninja -j8

I also tried this by passing --compiler=g++-5/g++-6/
/opt/scylladb/bin/g++-7 and --enable-dpdk, but every time, I get the same error:

[63/63] AR build/release/libseastar.a
ninja: build stopped: subcommand failed.

Ninja output: [error.log](https://github.com/scylladb/scylla/files/1535334/error.log)

**Comment by avikivity** [comment]
The error log shows you are using gcc 5.

**Comment by aliireza** [comment]
Yes. I also tried with gcc 6.3 and gcc 7.2. This time, I got some errors from the boost library, but I already have the libboost-all-dev package. Which gcc version should I use? 

[error2.log](https://github.com/scylladb/scylla/files/1535997/error2.log)
[error3.log](https://github.com/scylladb/scylla/files/1536070/error3.log)

**Comment by avikivity** [comment]
Try installing the boost version provided by our ppa, and adding `--cflags='-I/opt/scylladb/include -L/opt/scylladb/lib64'` to `./configure.py`.

**Comment by aliireza** [comment]
I installed every package from the Scylla's ppa and it solved my problem. Thank you very much for your help.

**Comment by jxingpz** [comment]
I got similar issue. Could you share some details how to fix this issue? Thanks,

**Comment by aliireza** [comment]
I did as follows:

```
git clone https://github.com/scylladb/scylla
cd scylla
git submodule update --init --recursive
sudo ./install_dependencies.sh
```
If you get an error about the authentication of Scylladb repositories, do as follows and then run the `sudo ./install_dependencies.sh` again.

```
curl https://s3.amazonaws.com/downloads.scylladb.com/deb/unstable/scylladb.gpg.pubkey|sudo apt-key add -
cd /etc/apt/sources.list.d/
sudo curl -O https://s3.amazonaws.com/downloads.scylladb.com/deb/unstable/xenial/master/6/scylla.list
sudo apt-get update
```

Install all of the packages from the scylla repositories:
```
sudo apt-get install $(grep -vE "^\s*#" packages.txt  | tr "\n" " ")
```

- [packages.txt](https://github.com/scylladb/scylla/files/1557448/packages.txt)

Configure ninja.build:

```
./configure.py --mode=all --dpdk-enable --cflags='-I/opt/scylladb/include -L/opt/scylladb/lib64' --compiler=/opt/scylladb/bin/g++-7
```
You can also use gcc 6.x/7.x instead of passing `--compiler=/opt/scylladb/bin/g++-7` option to `configure.py`. 

Link: https://askubuntu.com/questions/466651/how-do-i-use-the-latest-gcc-on-ubuntu

Build:

```
ninja -j8
```

**Comment by jxingpz** [comment]
Thanks for the quick  response, but, I get below error on step: "sudo apt-get update"

root@vizion-build-vm-102:/etc/apt/sources.list.d# sudo apt-get update
Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:2 http://downloads.scylladb.com/deb/3rdparty/xenial xenial InRelease
Hit:3 http://ppa.launchpad.net/webupd8team/java/ubuntu xenial InRelease
Hit:4 http://downloads.scylladb.com/deb/unstable/xenial/master/6 xenial InRelease
Hit:5 https://deb.nodesource.com/node_8.x xenial InRelease
Err:6 https://dl.yarnpkg.com/debian stable InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY E074D16EB6FF4DE3
Hit:7 http://us.archive.ubuntu.com/ubuntu xenial InRelease
Hit:8 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:9 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease
Reading package lists... Done
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://dl.yarnpkg.com/debian stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY E074D16EB6FF4DE3
W: Failed to fetch https://dl.yarnpkg.com/debian/dists/stable/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY E074D16EB6FF4DE3
W: Some index files failed to download. They have been ignored, or old ones used instead.


=== after adding this ===
 sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E074D16EB6FF4DE3
===============

I am able go forward, but, I got more error for install

root@vizion-build-vm-102:/etc/apt/sources.list.d#  sudo apt-get install scylla-antlr35-c++-dev
Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package scylla-antlr35-c++-dev
E: Couldn't find any package by regex 'scylla-antlr35-c++-dev'


And, many similar errors for other modules. 
......................
Reading state information... Done
E: Unable to locate package scylla-antlr35-c++-dev
E: Couldn't find any package by regex 'scylla-antlr35-c++-dev'
E: Unable to locate package scylla-env
E: Unable to locate package scylla-libboost163-all-dev
E: Unable to locate package scylla-libboost163-dev
E: Unable to locate package scylla-libboost163-doc
E: Unable to locate package scylla-libboost163-tools-dev
E: Unable to locate package scylla-libboost-atomic163.0
E: Couldn't find any package by glob 'scylla-libboost-atomic163.0'
E: Couldn't find any package by regex 'scylla-libboost-atomic163.0'
E: Unable to locate package scylla-libboost-atomic163-dev
E: Unable to locate package scylla-libboost-chrono163.0
E: Couldn't find any package by glob 'scylla-libboost-chrono163.0'
E: Couldn't find any package by regex 'scylla-libboost-chrono163.0'
E: Unable to locate package scylla-libboost-chrono163-dev
E: Unable to locate package scylla-libboost-context163.0
E: Couldn't find any package by glob 'scylla-libboost-context163.0'
........................

Thanks again.

**Comment by aliireza** [comment]
Add this line to the end of `/etc/apt/source.list` and run `sudo apt-get update`:

`deb http://ppa.launchpad.net/scylladb/ppa/ubuntu xenial main
`

then try again.
